snippets/pyaloha/main.py

```diff
@@ -74,44 +74,63 @@ def aggregate_raw_data(
     setup_logs()
     logger = multiprocessing.get_logger()
 
-    pool = multiprocessing.Pool(worker_num)
+    files = [
+        os.path.join(data_dir, fname)
+        for fname in sorted(os.listdir(data_dir))
+        if check_fname(fname, start_date, end_date)
+    ]
 
-    try:
-        files = [
-            os.path.join(data_dir, fname)
-            for fname in os.listdir(data_dir)
-            if check_fname(fname, start_date, end_date)
-        ]
-
-        items = (
-            (plugin_dir, plugin, fpath, events_limit)
-            for fpath in files
-        )
+    tasks = [
+        (plugin_dir, plugin, fpath, events_limit)
+        for fpath in files
+    ]
 
-        aggregator = load_plugin(
-            plugin, plugin_dir=plugin_dir
-        ).DataAggregator(results_dir)
-
-        logger.info('Aggregator: aggregate')
-        for i, results in enumerate(pool.imap(invoke_cmd_worker, items)):
-            try:
-                aggregator.aggregate(WorkerResults.loads_object(results))
-            except Exception:
-                logger.error(
-                    'Aggregator: processing of %s failed: %s' % (
-                        files[i],
-                        traceback.format_exc()
-                    )
+    aggregator = load_plugin(
+        plugin, plugin_dir=plugin_dir
+    ).DataAggregator(results_dir)
+
+    logger.info('Aggregator: start workers')
+
+    pool = multiprocessing.Pool(worker_num, maxtasksperchild=None)
+    try:
+        engine = pool.imap_unordered
+        batch_size = 2 * worker_num
+        batch_number = len(tasks) / batch_size
+        for batch_no in range(batch_number + 1):
```

Comment:
```
batch_number = (len(tasks) + batch_size - 1) / batch_size
for batch_no in range(batch_number):
```

Or at least
```
batch_number = len(tasks) / batch_size + 1
for batch_no in range(batch_number):
```

Reply:
from __future__ import division
import math
...
batch_number = math.ceil(len(tasks) / batch_size) ?

Reply:
Also possible, although this way you introduce intermediate floats.

Reply:
done